sbi/src/api/tsm.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::TeeFunction::*; // TODO
+use crate::{ecall_send, Error, Result, SbiMessage};
+use crate::{TsmInfo, TsmPageType, TvmCpuRegister, TvmCreateParams}; //TODO
+
+/// Initiates a TSM fence on this CPU.
+pub fn initiate_fence() -> Result<()> {
+    let msg = SbiMessage::Tee(TsmInitiateFence);
+    // Safety: TsmInitiateFence doesn't read or write any memory we have access to.
+    unsafe { ecall_send(&msg).map(|_| ()) }
+}
+
+/// Returns information about the TSM that booted this host context.
+pub fn get_info() -> Result<TsmInfo> {
+    let mut tsm_info = TsmInfo::default();
+    let tsm_info_size = core::mem::size_of::<TsmInfo>() as u64;
+    let msg = SbiMessage::Tee(TsmGetInfo {
+        dest_addr: &mut tsm_info as *mut _ as u64,
+        len: tsm_info_size,
+    });
+    // Safety: The passed info pointer is uniquely owned so it's safe to modify in SBI.
+    let tsm_info_len = unsafe { ecall_send(&msg).expect("TsmGetInfo failed") };
```

Comment:
Propagate the error instead of `expect()`ing?

Reply:
done